opts/envdir.go

```diff
@@ -0,0 +1,43 @@
+package opts
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
+/*
+Read in a directory with files acting as environment variables.
+*/
+func ParseEnvDir(directory string) ([]string, error) {
+	dir, err := ioutil.ReadDir(directory)
+	if err != nil {
+		return []string{}, err
+	}
+
+	lines := []string{}
+	for _, info := range dir {
+		variable := info.Name()
+		// ignore directories and hidden files
+		if !info.IsDir() && variable[0] != '.' {
+			if strings.ContainsAny(variable, whiteSpaces) {
+				return []string{}, ErrBadEnvVariable{fmt.Sprintf("variable '%s' has white spaces", variable)}
+			}
+
+			data, err := ioutil.ReadFile(filepath.Join(directory, variable))
```

Comment:
Disagree with `ReadFile` here: the [`envdir`](http://cr.yp.to/daemontools/envdir.html) page mentions it only reads the **first-line**:

> If  d contains a file named s whose **first line** is t [...]